extensions/sql-migration/src/dialog/skuRecommendationResults/SelectStorageAccountDialog.ts

```diff
@@ -0,0 +1,608 @@
+		try {
+			this._azureStorageAccountDropdown.loading = true;
+
+			this._storageAccounts = await utils.getStorageAccounts(
+				this._azureAccount,
+				this._targetSubscription
+			);
+
+
+			this._azureStorageAccountDropdown.values = utils.getAzureResourceDropdownValues(
+				this._storageAccounts,
+				this._location!,
+				this._resourceGroup?.name,
+				constants.NO_STORAGE_ACCOUNT_FOUND);
+
+			utils.selectDefaultDropdownValue(
+				this._azureStorageAccountDropdown
+			);
+
+			const selectedBlobStorage = this._storageAccounts?.find(rg => rg.name === (<azdata.CategoryValue>this._azureStorageAccountDropdown.value)?.displayName);
+			this._storageAccount = (selectedBlobStorage)
+				? utils.deepClone(selectedBlobStorage)!
+				: undefined!;
+
+		} catch (error) {
+			logError(TelemetryViews.UploadArmTemplateDialog, '_populateStorageAccount', error);
+			void vscode.window.showErrorMessage(
+				constants.SELECT_SERVICE_ERROR,
+				error.message);
+		} finally {
+			this._azureStorageAccountDropdown.loading = false;
+		}
+	}
+
+	private async _populateBlobContainer(): Promise<void> {
+		try {
+			this._dialog.okButton.enabled = false;
+			this._blobContainerDropdown.loading = true;
+
+			this._blobContainers = await utils.getBlobContainer(this._azureAccount, this._targetSubscription, this._storageAccount);
+
+			this._blobContainerDropdown.values = utils.getResourceDropdownValues(
+				this._blobContainers,
+				constants.NO_BLOBCONTAINERS_FOUND);
+
+			utils.selectDefaultDropdownValue(
+				this._blobContainerDropdown
+			);
+
+			const selectedBlobContainer = this._blobContainers?.find(rg => rg.name === (<azdata.CategoryValue>this._blobContainerDropdown.value)?.displayName);
+			this._blobContainer = (selectedBlobContainer)
+				? utils.deepClone(selectedBlobContainer)!
+				: undefined!;
+
+			if ((<azdata.CategoryValue>this._blobContainerDropdown.value)?.displayName !== constants.NO_BLOBCONTAINERS_FOUND) {
+				this._dialog.okButton.enabled = true;
+			}
+
+
+		} catch (error) {
+			logError(TelemetryViews.UploadArmTemplateDialog, '_populateBlobContainer', error);
+			void vscode.window.showErrorMessage(
+				constants.SELECT_SERVICE_ERROR,
+				error.message);
+		} finally {
+			this._blobContainerDropdown.loading = false;
+		}
+	}
+
+	private async uploadTemplate(): Promise<void> {
+		const storageKeys = await getStorageAccountAccessKeys(this._azureAccount, this._targetSubscription, this._storageAccount);
+		const accountName = this._storageAccount.name;
+		const containerName = this._blobContainer.name;
+		const date = new Date().toISOString().slice(0, 10);
+		const blobName = `ARMTemplate-${this._targetType}-${date}.json`;
+
+		const sharedKeyCredential = new StorageSharedKeyCredential(this._storageAccount.name, storageKeys.keyName1);
+
+		const sasToken = generateBlobSASQueryParameters({
+			containerName,
+			blobName,
+			permissions: BlobSASPermissions.parse("racwd"),
+			expiresOn: new Date(new Date().valueOf() + 86400),
```

Comment:
are we expecting 24 Hours expiration?

Reply:
I just used some random value, can change based on suggestion.